web/modules/custom/adms_validator/src/Form/AdmsValidatorForm.php

```diff
@@ -0,0 +1,224 @@
+<?php
+
+namespace Drupal\adms_validator\Form;
+
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\file\FileInterface;
+use Drupal\rdf_entity\Database\Driver\sparql\Connection;
+use EasyRdf\Graph;
+use EasyRdf\GraphStore;
+use EasyRdf\Sparql\Result;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Form to validate the ADMS compliance of a RDF or TTL file.
+ */
+class AdmsValidatorForm extends FormBase {
+
+  /**
+   * The name of the graph used for validation.
+   *
+   * @var string
+   */
+  const VALIDATION_GRAPH = 'http://adms-validator/';
+
+  /**
+   * The path of the file that contains the validation rules.
+   *
+   * @var string
+   */
+  const SEMIC_VALIDATION_QUERY_PATH = "SEMICeu/adms-ap_validator/python-rule-generator/ADMS-AP Rules .txt";
+
+  /**
+   * The Sparql endpoint.
+   *
+   * @var \Drupal\rdf_entity\Database\Driver\sparql\Connection
+   */
+  protected $sparqlendpoint;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('sparql_endpoint')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @param \Drupal\rdf_entity\Database\Driver\sparql\Connection $sparql_endpoint
+   *   The Sparql endpoint.
+   */
+  public function __construct(Connection $sparql_endpoint) {
+    $this->sparqlendpoint = $sparql_endpoint;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'adms_validator_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $form['adms_file'] = [
+      '#type' => 'file',
+      '#title' => $this->t('File'),
+      '#description' => $this->t('An RDF file you want to test for compliance.'),
+      '#upload_validators'  => [
+        'file_validate_extensions' => ['rdf ttl'],
+      ],
+    ];
+    $form['info'] = [
+      '#markup' => $this->t('This validator uses the <a href="https://github.com/SEMICeu/adms-ap_validator">SEMIC ADMS-AP ruleset</a>.'),
+    ];
+    $form['submit'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Upload'),
+      '#button_type' => 'primary',
+    ];
+    $info = $form_state->getBuildInfo();
+    if (!empty($info['validation_errors'])) {
```

Comment:
Let's add some messages to the user when there are no validation errors, as we discussed.